docs/reference-guides/block-api/block-patterns.md

```diff
@@ -135,4 +136,48 @@ function my_plugin_unregister_my_pattern_categories() {
 }
  
 add_action( 'init', 'my_plugin_unregister_my_pattern_categories' );
-```
\ No newline at end of file
+```
+
+## Block patterns contextual to block types
+
+It is possible to attach a block pattern to one or more block types. This adds the block pattern as an available transform for that block type.
+
+For instance:
+
+```php
+register_block_pattern(
+	'my-plugin/powered-by-wordpress',
+	array(
+		'title'      => __( 'Powered by WordPress', 'my-plugin' ),
+		'blockTypes' => array( 'core/paragraph' ),
+		'content'    => '<!-- wp:paragraph -->
+		<p>Powered by WordPress</p>
+		<!-- /wp:paragraph -->
+		',
+	)
+);
+```
+
+The above code registers a block pattern named 'my-plugin/powered-by-wordpress' and also shows the pattern in the "transform menu" of paragraph blocks.
+
+Blocks can also use these contextual block patterns in other places. For instance, when inserting a new Query Loop block, the user is provided with a list of all patterns attached to the block.
+
+## Semantic block patterns
+
+In block themes, you can also mark block patterns as "header" or "footer" patterns (template part areas). We call these "semantic block patterns". These patterns are shown to the user when inserting or replacing header or footer template parts.
+
+Example:
+
+```php
+<?php
+register_block_pattern(
+	'my-plugin/my-header',
+	array(
+		'title'      => __( 'My Header', 'my-plugin' ),
+		'categories' => array( 'header' ),
+		// Assigning the pattern the "header" area.
+		'blockTypes' => array( 'core/template-part/header' ),
```

Comment:
It's worth noting that `core/template-part/header` is something that is used only by `template parts` and is not a generic API for block variations. It had been discussed to support block variations with the same format, but eventually left for template parts to override the filtering patterns function with that naming convention.